react-app-drawer-component/dist/styles.css

```diff
@@ -0,0 +1,91 @@
+* {
+  box-sizing: border-box;
+}
+
+body {
+  margin: 0;
+  font-family: sans-serif;
+}
+
+.row {
+  display: flex;
+}
+
+.col-100 {
+  flex-basis: 100%;
+}
+
+.col-80 {
+  flex-basis: 80%;
+}
+
+.col-20 {
+  flex-basis: 20%;
+}
+
+.overlay-open,
+.overlay-closed {
+  position: absolute;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  z-index: 1;
+}
+
+.overlay-open {
+  background-color: rgb(0 0 0 / 75%);
+  left: 0;
+  transition: 0.4s ease-in;
+}
+
+.overlay-closed {
+  background-color: rgb(0 0 0 / 0%);
+  left: -100%;
+  transition: 0.4s ease-in;
+}
+
+.drawer-open,
+.drawer-closed {
+  position: fixed;
+  background-color: #fff;
+  height: 100%;
+  width: 200px;
+  z-index: 2;
+}
+
+.drawer-open {
+  left: 0;
+  transition: 0.4s ease-in;
+}
+
+.drawer-closed {
+  left: -200px;
```

Comment:
always specify which CSS property you are trying to transition. the default is `all`, which doesn't really convey your real intention.

https://developer.mozilla.org/en-US/docs/Web/CSS/transition#syntax

also something is weird. like, the width of the drawer shouldn't change at any point, but its box offset should.